### ISSUE TITLE
[pdf_viewer] Provide an onTap callback

### ISSUE THREAD

**Issue by tpouriel** [syncfusion/flutter-widgets#1260]
All provided solutions on issues about GestureDetector wrapping the viewer widget feels hacky because there must be only one winner for the GestureArena.
I even got the same error than in issue #1231 on scaling.

Why can't you just expose a callback when a page is tapped ?

**Comment by manojSF3414** [comment]
We checked the usage of RawGestureDetector on our PDF Viewer, and it works fine for us. We have prepared a sample for your reference, and that can be downloaded from the attachment. Please try this and revert us if you still face the issue or not.

Also, we would like to know your exact requirement (use case) for needing the GestureDectector or the onTap callback. This will help us more on analyzing things more detailed way and provide you with a proper solution at the earliest.

[boldesk_473860.zip](https://github.com/syncfusion/flutter-widgets/files/11767216/boldesk_473860.zip)

**Comment by deepika134** [comment]
We have considered the requirement to provide onTap callback in SfPdfViewer and this support will be included in our upcoming 2023 Volume 2 - Service Pack release which is expected to be available on this July end. Kindly use the following link for track the status of the feature,
https://www.syncfusion.com/feedback/44761/support-for-ontap-callback

**Comment by deepika134** [comment]
We are glad to announce that the support for onTap callback is available from the Essential Studio 2023 Volume 2 Service Pack v22.2.5 release. Please find the download link below,

[Essential Studio 2023 Volume 2 Service Pack Release v22.2.5 is available for download | Announcements Forums | Syncfusion](https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.syncfusion.com%2Fforums%2F183720%2Fessential-studio-2023-volume-2-service-pack-release-v22-2-5-is-available-for-download&data=05%7C01%7Cdeepika.ravi%40syncfusion.com%7C52e3cac7ca75411d7aa508db8f29de2c%7C77f1fe12b04949198c509fb41e5bb63b%7C0%7C0%7C638261182219128270%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=qsP518%2B0TgnBns9l7yYnum8eoLJJXQrzWortLiA1tBM%3D&reserved=0)

Package link: https://pub.dev/packages/syncfusion_flutter_pdfviewer/versions/22.2.5 

We thank you for your support and appreciate your patience in waiting for this release. Please get in touch with us if you would require any further assistance.

**Comment by tpouriel** [comment]
It does work like a charm, thank you for your time and consideration !

**Comment by abhinavsinghring** [comment]
thank you for adding onTap callback 
but when I'm double tapping to zoom onTap function is also getiing executed

```
SfPdfViewer.file(
  File(widget.url),
    onTap: (details) {
      if(isFulScreen==false){
        SystemChrome.setEnabledSystemUIMode(SystemUiMode.immersiveSticky).then((value) {
          setState(() {
            isFulScreen = true;
          });
        });
      }
      if(isFulScreen==true){
        SystemChrome.setEnabledSystemUIMode(SystemUiMode.edgeToEdge).then((value) {
          setState(() {
            isFulScreen = false;
          });
        });
      }
    },
    maxZoomLevel: 4,
    onHyperlinkClicked: (details) async {
      final url = Uri.parse(details.uri);
      if (await canLaunchUrl(url)) {
        await launchUrl(url, mode: LaunchMode.externalApplication);
      }
    },
  )
```

**Comment by ramkumarsync** [comment]
The support for the **onTap** callback to retrieve the page number, position, and page position when tapping on the SfPdfViewer is available from version v22.2.5. Please find the package link below.

https://pub.dev/packages/syncfusion_flutter_pdfviewer/versions/22.2.5
